ui/src/SyntaxResult.js

```diff
@@ -61,7 +69,8 @@ export default function SyntaxResult({ content, status }) {
                             <td>{item.column}</td>
                             <td>
                               <span class='pre'>{item.msg.split('\n').slice(0, -2).join('\n')}</span>
-                              <span class='pre mono'>{item.msg.split('\n').slice(-2).join('\n')}</span>
+                              <br /> {}
+                              <span class='pre mono' dangerouslySetInnerHTML={{ __html: modifiedStr }}></span>
```

Comment:
I don't think `dangerouslySetInnerHTML` is needed here? It's actually risky in this case.

Because the IFC could be

#1=IFCPERSON($, $, '<script>alert("haha")</script>'

And in that case the text that was embedded in the IFC content will be executed as actual HTML. This vulnerability is called XSS for cross site scripting and could potentially result in an attacker having access to your account.

Reply:
Right, it doesn't even seem that difficult to embed some malicious html.

I think the following is safer. The message is split into three parts (in which the caret `^`is the highlighted part) and rendered seperately as a react (Fragment) component. 
```
  function renderMessage(error_instance, numWhitespace, numHighlight) {
    const beforeHighlight = error_instance.substring(0, numWhitespace);
    const highlightedChar = error_instance.substring(numWhitespace, numWhitespace + numHighlight);
    const afterHighlight = error_instance.substring(numWhitespace + numHighlight);
    return (
      <React.Fragment>
        {beforeHighlight}
        <span
          style={{
            textDecoration: 'underline',
            fontWeight: 'bold',
            backgroundColor: '#ddd',
            border: 'solid 1px red'
          }}
        >
          {highlightedChar}
        </span>
        {afterHighlight}
      </React.Fragment>
    );
  };
```
Define const
```
              const errorMessage = renderMessage(
                error_instance,
                numWhitespace,
                numHighlight
              );
```
Display table
`<span class='pre mono'>{errorMessage}</span>`